File: branches/rev5236/support/MacOSX/pgplot/pgplot-src-mac/sys_vms/grgmem.f
C Fortran callable memory allocator (OpenVMS)
C
C Called as :
C	ier = grgmem (size,pointer)
C
C where : size is an integer size of memory to allocate
C	  pointer is an integer to return the pointer into

      INTEGER FUNCTION GRGMEM(SIZE, POINTER)
      INTEGER SIZE, POINTER
      INTEGER LIB$GET_VM
      GRGMEM = LIB$GET_VM(SIZE, POINTER)
      END

C Fortran callable memory deallocator
C
C Called as :
C	ier = grfmem (size,pointer)
C
C where : size is an integer size of memory to deallocate (not used)
C	  pointer is an integer that contains the pointer

      INTEGER FUNCTION GRFMEM(SIZE, POINTER)
      INTEGER SIZE, POINTER
      INTEGER LIB$FREE_VM
      GRFMEM = LIB$FREE_VM(SIZE, POINTER)
      END

File: branches/mcxtrace-1.2/support/MacOSX/pgplot/pgplot-src-mac/sys_cray/grgmem.c
#include <stdio.h>
#include <stdlib.h>
/*
  Fortran callable memory allocator (Cray version).

  Called as :
	ier = grgmem (size,pointer)

  where : size is an integer size of memory to allocate
	  pointer is an integer to return the pointer into

*/

int GRGMEM(size, pointer)
int *size, *pointer;
{
  char *area = malloc(*size);
  *pointer = (int)area;
  if (area == NULL) return 0;
  return 1;
}

/*
  Fortran callable memory deallocator

  Called as :
	ier = grfmem (size,pointer)

  where : size is an integer size of memory to deallocate (not used)
	  pointer is an integer that contains the pointer

*/

int GRFMEM(size, pointer)
int *size, *pointer;
{
  char *area = (char *)*pointer;
  free(area);
  return 1;
}
